internal/infrastructure/kubernetes/infra.go

```diff
@@ -47,6 +47,9 @@ type Infra struct {
 	// Namespace is the Namespace used for managed infra.
 	Namespace string
 
+	// ControllerNamespace is the Namespace used for Envoy Gateway controller.
```

Comment:
shouldnt this be available in Runner Config  ?

Reply:
Runner config calls the NewInfra, I'm adding this field to be able to differentiate the namespace used for Infra and Controller

Reply:
referring to https://github.com/envoyproxy/gateway/blob/ef507181e89219c04faf7fbe67e4bcce898b751e/internal/infrastructure/runner/runner.go#L27

Reply:
controller namespace will be fetched from the Config

Reply:
so, we dont have access to the runner context anymore ? is that the issue ?

Reply:
@arkodg runner context is not in the Infra, instead, only the needed fields are passed down to the Infra structure.

```
func NewInfra(cli client.Client, cfg *config.Server) *Infra {
	var infraNamespace string
	if !cfg.EnvoyGateway.GatewayNamespaceMode() {
		infraNamespace = cfg.ControllerNamespace
	}
	return &Infra{
		Namespace:           infraNamespace,
		ControllerNamespace: cfg.ControllerNamespace,    # cfg is runner config
		DNSDomain:           cfg.DNSDomain,
		EnvoyGateway:        cfg.EnvoyGateway,
		Client:              New(cli),
		logger:              cfg.Logger.WithName(string(egv1a1.LogComponentInfrastructureRunner)),
	}
}
```